src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/edgpsi/experimental_detectron_generate_proposals_single_image_kernel_ref.cpp

```diff
@@ -0,0 +1,176 @@
+        return false;
+    }
+    return true;
+}
+
+namespace {
+constexpr size_t kImInfoInputIdx = 0;
+constexpr size_t kAnchorsInputIdx = 1;
+constexpr size_t kDeltasInputIdx = 2;
+constexpr size_t kScoresInputIdx = 3;
+constexpr size_t kRoiScoresOutputIdx = 4;
+
+ExperimentalDetectronGenerateProposalsSingleImageRef::DispatchData SetDefault(
+        const experimental_detectron_generate_proposals_single_image_params& params, size_t idx) {
+    ExperimentalDetectronGenerateProposalsSingleImageRef::DispatchData dispatch_data;
+
+    if (idx == 0) {
+        const auto bottom_H = params.inputs[kDeltasInputIdx].Feature().v;
+        const auto bottom_W = params.inputs[kDeltasInputIdx].Y().v;
+        const auto anchors_num = params.inputs[kScoresInputIdx].Batch().v;
+        dispatch_data.gws = {bottom_H, bottom_W, anchors_num};
+    } else if (idx == 3) {
+        dispatch_data.gws = {params.post_nms_count, 1, 1};
+    } else {
+        dispatch_data.gws = {1, 1, 1};
+    }
+    dispatch_data.lws = GetOptimalLocalWorkGroupSizes(dispatch_data.gws, params.engineInfo);
+
+    return dispatch_data;
+}
+}  // namespace
+
+void ExperimentalDetectronGenerateProposalsSingleImageRef::SetKernelArguments(
+        const experimental_detectron_generate_proposals_single_image_params& params,
+        size_t idx, cldnn::arguments_desc& arguments) const {
+    switch (idx) {
+        case 0: { // refine anchors
+            arguments.push_back({ArgumentDescriptor::Types::INPUT, kImInfoInputIdx});
+            arguments.push_back({ArgumentDescriptor::Types::INPUT, kAnchorsInputIdx});
+            arguments.push_back({ArgumentDescriptor::Types::INPUT, kDeltasInputIdx});
+            arguments.push_back({ArgumentDescriptor::Types::INPUT, kScoresInputIdx});
+            arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0}); // proposals
+            break;
+        }
+        case 1: { // sort proposals by score
+            arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0}); // proposals
+            break;
+        }
+        case 2: { // NMS
+            arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0}); // proposals
+            arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 1}); // nms_out_indices
+            arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 2}); // nms_num_outputs
+            break;
+        }
+        case 3: { // convert proposals to rois and roi scores
+            arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 0}); // proposals
+            arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 1}); // nms_out_indices
+            arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 2}); // nms_num_outputs
+            arguments.push_back({ArgumentDescriptor::Types::OUTPUT, 0});          // rois
+            arguments.push_back({ArgumentDescriptor::Types::INPUT, kRoiScoresOutputIdx}); // roi scores
```

Comment:
Could you please explain why roi scores argument is marked as INPUT and not OUTPUT?

Reply:
no, here should be INPUT, as we return indices as input mutable data